### PR TITLE
#14803 pull and show contents with publish date in TM

### PR DESCRIPTION
**Pull request by erickgonzalez**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/directive/ContentletDetail.java [hunk 1]
```diff
@@ -58,10 +59,18 @@ else if(type.baseType() == BaseContentType.WIDGET  && !Config.getBooleanProperty
       throw new ResourceNotFoundException("cannnot find contentlet id " +  argument + " lang:" + params.language);
     }
 
+    // If the time machine date is after the Publish date of the Contentlet (identifier data)
+    // we need to show the working. If not we only show the live.
+
+    boolean showWorking = false;
+
+    if (context.get("_show_working_") != null && (boolean)context.get("_show_working_")) {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/directive/ContentletDetail.java]: you can do something like
```
final boolean showWorking = ConversionUtils.toBoolean("_show_working_", context, false);
```
Also "_show_working_" should be a contanst

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/directive/ContentletDetail.java [hunk 2]
```diff
@@ -58,10 +59,18 @@ else if(type.baseType() == BaseContentType.WIDGET  && !Config.getBooleanProperty
...
+      showWorking = true;
+    }
 
 
 
-      return  File.separator +  params.mode.name() + File.separator   + argument + "_" + cv.getLang() + "." + EXTENSION;
+      return  File.separator + ( (!showWorking) ? params.mode.name() : PageMode.PREVIEW_MODE.name() ) + File.separator   + argument + "_" + cv.getLang() + "." + EXTENSION;
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/directive/ContentletDetail.java]: 1) You can use StringBuilder here
2) replace "_" by StringPool.UNDERLINE
3) replace "." by  StringPool.PERIOD
4) Agree parentheses are useless for showWorking

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/services/PageContextBuilder.java [hunk 3]
```diff
@@ -160,8 +160,14 @@ private void populateContainers() throws DotDataException, DotSecurityException
         final Language language = request == null ? APILocator.getLanguageAPI().getDefaultLanguage() :
                 WebAPILocator.getLanguageWebAPI().getLanguage(request);
 
+        boolean live = mode.showLive;
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/services/PageContextBuilder.java]: you can replace it by
```
final boolean live = (request.getSession(false) != null && request.getSession().getAttribute("tm_date")!=null)?false:mode.showLive;
```

Comment (erickgonzalez) [reply dotCMS/src/main/java/com/dotcms/rendering/velocity/services/PageContextBuilder.java]: done

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/services/PageContextBuilder.java [hunk 4]
```diff
@@ -171,7 +177,7 @@ private void populateContainers() throws DotDataException, DotSecurityException
                     final Set<String> cons = pageContents.get(containerId, uniqueId);
 
                     final User systemUser = APILocator.getUserAPI().getSystemUser();
-                    final Container c = (mode.showLive) ? (Container) APILocator.getVersionableAPI()
+                    final Container c = (live) ? (Container) APILocator.getVersionableAPI()
```
[the remark this answers is not in the document]
Comment (erickgonzalez) [reply dotCMS/src/main/java/com/dotcms/rendering/velocity/services/PageContextBuilder.java]: done

[the remark this answers is not in the document]
Comment (erickgonzalez) [reply dotCMS/src/main/java/com/dotcms/rendering/velocity/services/PageContextBuilder.java]: done

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/directive/ContentletDetail.java (hunk 2 above)
Comment (erickgonzalez) [reply dotCMS/src/main/java/com/dotcms/rendering/velocity/directive/ContentletDetail.java]: done

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/directive/ContentletDetail.java (hunk 1 above)
Comment (erickgonzalez) [reply dotCMS/src/main/java/com/dotcms/rendering/velocity/directive/ContentletDetail.java]: done

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/directive/ContentletDetail.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (erickgonzalez) [reply dotCMS/src/main/java/com/dotcms/rendering/velocity/directive/ContentletDetail.java]: done

**Review by nollymar — APPROVED** [review]

**Review by dsilvam — APPROVED** [review]

**Pull request merged**